### PR TITLE
Handling external URLs as profile pictures

### PR DESCRIPTION
**Pull request by tomasz-ludek**
## Screenshots / Videos

ezgif-7-c67fc7446f6b

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

**Comment by HashEngineering** [comment]
```
=== stack trace ===

com.bumptech.glide.load.engine.CallbackException: Unexpected exception thrown by non-Glide code
at com.bumptech.glide.load.engine.EngineJob.callCallbackOnLoadFailed(EngineJob.java:174)
at com.bumptech.glide.load.engine.EngineJob$CallLoadFailed.run(EngineJob.java:398)
at android.os.Handler.handleCallback(Handler.java:883)
at android.os.Handler.dispatchMessage(Handler.java:100)
at android.os.Looper.loop(Looper.java:237)
at android.app.ActivityThread.main(ActivityThread.java:8125)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: java.lang.IllegalStateException: Fragment ExternalUrlProfilePictureDialog{7fa088a} (a17fe6ba-7d0c-445c-aa3f-38111ce9ea08)} not attached to a context.
at androidx.fragment.app.Fragment.requireContext(Fragment.java:805)
at de.schildbach.wallet.ui.dashpay.ExternalUrlProfilePictureDialog$initCustomView$1$afterTextChanged$1.onLoadFailed(ExternalUrlProfilePictureDialog.kt:154)
at com.bumptech.glide.request.SingleRequest.setErrorPlaceholder(SingleRequest.java:425)
at com.bumptech.glide.request.SingleRequest.onLoadFailed(SingleRequest.java:683)
at com.bumptech.glide.request.SingleRequest.onLoadFailed(SingleRequest.java:643)
at com.bumptech.glide.load.engine.EngineJob.callCallbackOnLoadFailed(EngineJob.java:172)
... 8 more

Cause:

java.lang.IllegalStateException: Fragment ExternalUrlProfilePictureDialog{7fa088a} (a17fe6ba-7d0c-445c-aa3f-38111ce9ea08)} not attached to a context.
at androidx.fragment.app.Fragment.requireContext(Fragment.java:805)
at de.schildbach.wallet.ui.dashpay.ExternalUrlProfilePictureDialog$initCustomView$1$afterTextChanged$1.onLoadFailed(ExternalUrlProfilePictureDialog.kt:154)
at com.bumptech.glide.request.SingleRequest.setErrorPlaceholder(SingleRequest.java:425)
at com.bumptech.glide.request.SingleRequest.onLoadFailed(SingleRequest.java:683)
at com.bumptech.glide.request.SingleRequest.onLoadFailed(SingleRequest.java:643)
at com.bumptech.glide.load.engine.EngineJob.callCallbackOnLoadFailed(EngineJob.java:172)
at com.bumptech.glide.load.engine.EngineJob$CallLoadFailed.run(EngineJob.java:398)
at android.os.Handler.handleCallback(Handler.java:883)
at android.os.Handler.dispatchMessage(Handler.java:100)
at android.os.Looper.loop(Looper.java:237)
at android.app.ActivityThread.main(ActivityThread.java:8125)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
```

**Comment by tomasz-ludek** [comment]
The issue fixed https://github.com/dashevo/dash-wallet/pull/542/commits/61f78ab0075fd1ccd7c8878fc2da9fee0e095d68

File: wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt [hunk 1]
```diff
@@ -66,9 +63,13 @@ class EditProfileActivity : BaseMenuActivity() {
 
     private lateinit var editProfileViewModel: EditProfileViewModel
     private lateinit var selectProfilePictureSharedViewModel: SelectProfilePictureSharedViewModel
+    private lateinit var externalUrlSharedViewModel: ExternalUrlProfilePictureViewModel
+
     private var isEditing: Boolean = false
     private var defaultAvatar: TextDrawable? = null
 
+    private var prosilePictureChanged = false
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt:71]: `prosilePictureChanged` - **profile** is mispelled.

File: wallet/src/de/schildbach/wallet/ui/dashpay/CropImageActivity.kt [hunk 2]
```diff
@@ -41,25 +57,41 @@ class CropImageActivity : InteractionAwareActivity() {
...
-        Glide.with(this).load(tempFile).listener(object : RequestListener<Drawable> {
-            override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
-                Toast.makeText(this@CropImageActivity,
-                        R.string.unable_to_load_image, Toast.LENGTH_SHORT).show()
-                finish()
-                return false
-            }
+        Glide.with(this)
+                .load(tempFile)
+//                .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/dashpay/CropImageActivity.kt:62]: Is this commented line not needed?

File: wallet/src/de/schildbach/wallet/ui/dashpay/ExternalUrlProfilePictureDialog.kt [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+//                                return false
+//                            }
+//                            override fun onResourceReady(resource: Drawable?, model: Any?, target: Target<Drawable>?, dataSource: DataSource?, isFirstResource: Boolean): Boolean {
+//                                urlPreviewPane.visibility = View.VISIBLE
+//                                positiveButton.isEnabled = true
+//                                return false
+//                            }
+//                        })
+//                        .into(urlPreview)
+                .listener(object : RequestListener<Drawable> {
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/dashpay/ExternalUrlProfilePictureDialog.kt:173]: is this commented coded not needed?

**Review by HashEngineering — CHANGES_REQUESTED** [review]
There are some extra comments and one misspelled variable name.

File: wallet/src/de/schildbach/wallet/ui/dashpay/CropImageActivity.kt (hunk 2 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/dashpay/CropImageActivity.kt]: Removed

File: wallet/src/de/schildbach/wallet/ui/dashpay/ExternalUrlProfilePictureDialog.kt (hunk 3 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/dashpay/ExternalUrlProfilePictureDialog.kt]: Removed

File: wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt (hunk 1 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt]: Fixed

**Review by HashEngineering — APPROVED** [review]

File: wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt [hunk 4]
```diff
@@ -311,17 +333,51 @@ class EditProfileActivity : BaseMenuActivity() {
...
+            }
+        }
+        return newUriBuilder.build()
+    }
+
     private fun cropProfilePicture() {
         val tmpPictureUri = editProfileViewModel.tmpPictureFile.toUri()
         val profilePictureUri = editProfileViewModel.profilePictureFile!!.toUri()
-        val intent = CropImageActivity.createIntent(this, tmpPictureUri, profilePictureUri)
+        val initZoomedRect = ProfilePictureTransformation.extractZoomedRect(externalUrlSharedViewModel.externalUrl)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt:379]: `externalUrlSharedViewModel.externalUrl` is null when selecting a picture from Camera or Device.

Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt:379]: Fixed

**Pull request merged**